fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1620,6 +1620,13 @@ public void handleDMLStmt(ExecPlan execPlan, DmlStmt stmt) throws Exception {
                     sourceType,
                     context.getSessionVariable().getQueryTimeoutS());
 
+            // The metadata may be changed between plan() and beginTransaction().
+            // When to beginTransaction(), the plan is not ready and the tablet is dropped by balance.
+            // So we need to get txn first, to make the dropping tablet procedure to wait.
+            // This is the re-plan of the insert to fix the issue
+
+            execPlan = StatementPlanner.plan(stmt, context);
```

Comment:
could we only plan once?

Reply:
It's a good idea. Currently, we should place the begin_transaction() in front of any request. It needs much refactoring of the transaction framework. I have added a configuration for this.